visualizer/draw/bubbleprof-ui.js

```diff
@@ -353,12 +329,81 @@ class BubbleprofUI extends EventEmitter {
       }
       currentUI = currentUI.selectNode(layoutNode, animationQueue)
       if (layoutNode.node.uid === aggregateNode.uid) {
-        executeAnimationQueue(animationQueue)
+        animationQueue.execute()
         return currentUI
       }
     }
   }
 
+  setupHistory () {
+    // Mark the current history entry (on page load) as the initial one.
+    // There is a `window.history.length` property, but it includes the entries
+    // _before_ the current page (eg. the github issue that linked to this
+    // visualization) so we can't use it here.
+    window.history.replaceState({ initial: true }, null, window.location.hash || '')
+
+    this.on('navigation', ({ to, silent }) => {
+      // Only update history if this navigation was not caused by history.
+      if (!silent) {
+        this.pushHistory(to.getHash())
+      }
+
+      this.backBtn.isHidden = !this.hasHistoryEntries()
+      this.backBtn.draw()
+    })
+
+    window.addEventListener('popstate', (event) => {
+      const { hash } = event.state
```

Comment:
Hmm now that you mention the error I remember I got that when manually editing the hash. Did you also manually edit the hash or did it happen during standard browser history traversal?

Reply:
I don't think so, most times I changed the hash nothing seemed to happen. Not impossible because I had earlier tried changing the hash manually in that profile.